compositor_common/src/scene/builtin_transformations.rs

```diff
@@ -0,0 +1,22 @@
+use serde::{Deserialize, Serialize};
+
+use crate::util::RGBAColor;
+
+#[derive(Serialize, Deserialize, Clone)]
+#[serde(tag = "transformation", rename_all = "snake_case")]
+pub enum BuiltinTransformation {
+    TransformToResolution(TransformToResolution),
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+#[serde(tag = "strategy", content = "color", rename_all = "snake_case")]
+pub enum TransformToResolution {
+    /// Rescales input in both axis to match output resolution
+    Stretch,
+    /// Scales input preserving aspect ratio and cuts equal parts
+    /// from both sides in "sticking out" dimension
+    Fill,
+    /// Scales input preserving aspect ratio and
+    /// fill the rest of the texture with the provided color
+    Fit(RGBAColor),
```

Comment:
as we discussed offline, add align option (here or in follow up PR).
e.g. 
```
{
  "strategy": "fit",
  "background_color": "#FFFFFFFF"
  "align": "top" | "bottom" | "top-left" | "top-right" | "left" | "right"
 }
 ```
 e.g. top implies that it's centered horizontally
```

Reply:
I would prefer to do it this way in the follow up PR:
```
{
  "strategy": "fit",
  "background_color": "#FFFFFFFF",
  "align_x": "left" | "center" (default) | "right",
  "align_y": "top" | "center" (default) | "bottom",
 }
```